bigquery_etl/cli/query.py

```diff
@@ -506,76 +507,89 @@ def _backfill_query(
     run_checks,
 ):
     """Run a query backfill for a specific date."""
+    backfill_date_str = backfill_date.strftime("%Y-%m-%d")
+    if backfill_date_str in exclude:
+        click.echo(f"Skip {query_file_path} backfill for run date {backfill_date_str}")
+        return True
+
     project, dataset, table = extract_from_query_path(query_file_path)
 
     match partitioning_type:
         case PartitionType.DAY:
-            partition = backfill_date.strftime("%Y%m%d")
+            partition_date = backfill_date + timedelta(days=date_partition_offset)
+            partition = partition_date.strftime("%Y%m%d")
         case PartitionType.MONTH:
+            if date_partition_parameter != 0:
+                # TODO: Support offsets here e.g. desktop_mobile_search_clients_monthly_v1
+                raise ValueError("Unsupported offset for month partitions.")
             partition = backfill_date.strftime("%Y%m")
         case _:
             raise ValueError(f"Unsupported partitioning type: {partitioning_type}")
 
-    backfill_date = backfill_date.strftime("%Y-%m-%d")
-    if backfill_date not in exclude:
-        if destination_table is None:
-            destination_table = f"{project}.{dataset}.{table}"
+    if destination_table is None:
+        destination_table = f"{project}.{dataset}.{table}"
 
-        if not no_partition:
-            destination_table = f"{destination_table}${partition}"
+    if not no_partition:
+        destination_table = f"{destination_table}${partition}"
```

Comment:
I don't even understand this option anymore. We can only backfill partitioned tables, in what situation would we want a single date for the backfill to overwrite the entire table? Are they even backfilling other dates, or do they all overwrite the entire table?? Is it a backfill to init the table with a single date??? Then why not just write to that date's partition?????

Reply:
One use-case is rolling tables e.g. `firefox_android_clients_v1`. The output table might be partitioned or not but it's specifically not partitioned by the query parameter. e.g. a table partitioned by `first_seen_date` but that uses `submission_date` as a parameter. Generally it shouldn't be necessary, if you're running a backfill the metadata is probably enough to tell us how to run the query (it's enough to define the task in Airflow so)

Reply:
The problem is this doesn't generally support those incremental tables, just the partitioned ones. So it almost seems like it accidentally works for those tables

Reply:
Yes I agree, it works but in a roundabout way. The current assumption defaults to daily partition and `no_partition` will override that when deciding where to write. I'm actually in favour of removing `no_partition` and replacing it with just using metadata but that was out of scope for this PR